### PR TITLE
Fix $buttongroup-spacing

### LINKED ISSUE #8575: [Button Group] margin-bottom
If Button Groups have enough Buttons to wrap to multiple lines, the lines are flush against each other. Should we add `margin-bottom: $buttongroup-spacing`? 

Here's an example showing before/after this proposed fix: 
http://codepen.io/andycochran/pen/jqZYMX

Relatedly, how does this effect button radius (e.g. #8447 Should radius go on the button or the group)?

### LINKED ISSUE #8586: [Button Group] Stacked
We changed Button Groups from borders to margin for the `$buttongroup-spacing`, but `.stacked` is still using borders. 

screen shot 2016-04-12 at 12 25 43 pm
screen shot 2016-04-12 at 12 26 05 pm

### PR DESCRIPTION
**Pull request by andycochran**
This PR fixes some issues with button margins. 
- closes #8586 (`.stacked` was using borders for `$buttongroup-spacing`)
- closes #8575 (prevent buttons groups w/ multiple lines being flush against each other)
- appropriately uses rtl/ltr for `$buttongroup-spacing`
- simplifies the styles

### CODE REVIEW

**Comment by kball** [comment]
I love how much cleaner this ends up being.  @brettsmason @JeremyEnglert can one of you try this out and verify it works as described?

**Comment by brettsmason** [comment]
@andycochran I've had a look at this. The flex grid version works fine from what I can see. However on the expanded buttons the last item is dropping down a line (this seems to be the case on the current docs too).

**Comment by andycochran** [comment]
@brettsmason I can't seem to replicate the last item wrapping to the next line. Is this happening in a specific browser?

**Comment by brettsmason** [comment]
@andycochran It was in the latest version of Chrome on Windows 10. I've left the office for the day now so I'll confirm in the morning (I setup some visual tests to confirm everything was OK).

**Comment by brettsmason** [comment]
@andycochran Sorry it took a while to get back to you. I've just pulled down this again and checked at work, the last item is still dropping down. This happens in Chrome and in Edge. See below (first is this PR, 2nd is live docs).

button-group

f6-docs-bg

I haven't had time to see why this is yet, but I'll take another look when I'm home.

**Comment by andycochran** [comment]
@brettsmason Boo. I bet this is a [sub-pixel rounding](http://ejohn.org/blog/sub-pixel-problems-in-css/) bug. The `calc()` on the button width is probably rounding up a pixel. Any ideas?

**Comment by brettsmason** [comment]
I'll try and have a look over your code tonight if I get time and see if I can think of a solution.

**Comment by andycochran** [comment]
@brettsmason Maybe negative margin on the last child?

```scss
:last-child {
  margin-#{$global-right}: $buttongroup-spacing*-1;
}
```

**Comment by brettsmason** [comment]
@andycochran I think I've found the issue finally. Line 83 has this:
`@if #{$buttongroup-spacing} == '0' {` it should be `@if $buttongroup-spacing == '0' {`.
Changing the spacing value allowed me to notice the problem.

Take a look, I think thats whats causing the issue anyway...

**Comment by andycochran** [comment]
@brettsmason We actually don't need that `@if`. Could you see if this simplified logic helps? 

```scss
      @for $i from 2 through $buttongroup-expand-max {
        &:first-child:nth-last-child(#{$i}) {
          &, &:first-child:nth-last-child(#{$i}) ~ #{$selector} {
            display: inline-block;
            width: calc(#{percentage(1 / $i)} - #{$buttongroup-spacing});
            margin-#{$global-right}: $buttongroup-spacing;
            &:last-child {
              margin-#{$global-right}: $buttongroup-spacing * -$buttongroup-expand-max;
            }
          }
        }
      }

```

**Comment by brettsmason** [comment]
@andycochran I've tested that and it works perfectly. I tried up to 6 buttons with lots of spacing configurations in different browsers and all worked as it should. If you can update the PR and I'll merge it.

Good job!

**Comment by andycochran** [comment]
@brettsmason Should be good to go.

**Comment by brettsmason** [comment]
@andycochran great - I've pulled it down to check and all works perfectly.
Looks good to merge!

File: scss/components/_button-group.scss [hunk 1]
```diff
@@ -103,20 +104,12 @@ $buttongroup-expand-max: 6 !default;
...
-    
-    &:not(:last-child) {
-      @if $global-flexbox {
-        margin-bottom: $buttongroup-spacing;
-      }
-      @else {
-        border-bottom: $buttongroup-spacing solid $body-background;
-      }
+    &:last-child {
+      margin-bottom: 0;
```
Comment (DaSchTour) [inline scss/components/_button-group.scss]: I think this is the reason for #8780

**Pull request merged**